resources/qml/RoomSettings.qml

```diff
@@ -0,0 +1,268 @@
+    y: MainWindow.y + (MainWindow.height / 2) - (height / 2)
+    minimumWidth: 400
+    minimumHeight: 650
+    palette: colors
+    color: colors.window
+    modality: Qt.WindowModal
+
+    Shortcut {
+        sequence: StandardKey.Cancel
+        onActivated: roomSettingsDialog.close()
+    }
+
+    ColumnLayout {
+        id: contentLayout1
+
+        anchors.fill: parent
+        anchors.margins: 10
+        spacing: 10
+
+        Avatar {
+            url: roomSettings.roomAvatarUrl.replace("mxc://", "image://MxcImage/")
+            height: 130
+            width: 130
+            Layout.alignment: Qt.AlignHCenter
+            onClicked: {
+                if(roomSettings.canChangeAvatar) {
+                    roomSettings.updateAvatar();    
+                }
+            }
+        }
+
+        BusyIndicator {
+            Layout.alignment: Qt.AlignHCenter
+            running: roomSettings.isLoading
+            visible: roomSettings.isLoading
+        }
+
+        Text {
+            id: errorText
+            text: "Error Text"
+            color: "red"
+            visible: opacity > 0
+            opacity: 0
+            Layout.alignment: Qt.AlignHCenter
+        }
+
+        SequentialAnimation {
+            id: hideErrorAnimation
+            running: false
+            PauseAnimation {
+                duration: 4000
+            }
+            NumberAnimation {
+                target: errorText
+                property: 'opacity'
+                to: 0
+                duration: 1000
+            }
+        }
+
+        Connections{
+            target: roomSettings
+            onDisplayError: {
+                errorText.text = errorMessage
+                errorText.opacity = 1
+                hideErrorAnimation.restart()
+            }
+        }
+
+        ColumnLayout {
+            Layout.alignment: Qt.AlignHCenter
+
+            MatrixText {
+                text: roomSettings.roomName
+                font.pixelSize: 24
+                Layout.alignment: Qt.AlignHCenter
+            }
+
+            MatrixText {
+                text: "%1 member(s)".arg(roomSettings.memberCount)
+                Layout.alignment: Qt.AlignHCenter
+            }
+        }
+
+        ImageButton {
+            Layout.alignment: Qt.AlignHCenter
+            image: ":/icons/icons/ui/edit.png"
+            visible: roomSettings.canChangeNameAndTopic
+            onClicked: roomSettings.openEditModal()
+        }
+
+        ScrollView {
+            Layout.maximumHeight: 75
+            ScrollBar.horizontal.policy: ScrollBar.AlwaysOff
+            Layout.alignment: Qt.AlignHCenter
+            Layout.fillWidth: true
+
+            TextArea {
+                text: roomSettings.roomTopic
+                wrapMode: TextEdit.WordWrap
+                readOnly: true
+                background: null
+            }
+        }
+
+        MatrixText {
+            text: "SETTINGS"
+        }
+
+        RowLayout {
```

Comment:
You probably want to use a grid layout for these?

Reply:
Oh right yeah grid layout would be more suitable here